ContinousBenchmarking.md

```diff
@@ -0,0 +1,47 @@
+Continuous Benchmarking
+=======
+
+## Introduction:
+Benchmarking an algorithm is a compute intensive and time consuming chore that is essential for guaranteeing performance. 
+An algorithm can unknowingly drift towards an undesirable state if the laborious task of benchmarking is neglected for
+too long.  We envision a continous benchmarking system thats runs at least nightly, and posts results for the community to see.
+
+## Current State:
+Benchmarking is currently performed manually, by running a series of benchmarking scripts that can take up
+24 hours to complete.
+
+
+## Our current Tools:
+**Travis CI.** 
+    The current CI tool to manage the workflow of building garage and executing unit tests.  The Travis CI build 
+    environment have a maximum of 7.5 GB of memory, and 2 cores.  It also limits jobs to 120 minutes maximum for a
+    private Github repo, and 50 minutes for a public Github repo.  
+    https://docs.travis-ci.com/user/customizing-the-build/
+    https://docs.travis-ci.com/user/reference/overview/
+
+**Github Actions:**
+    Each workflow can run for up to 58 minutes (including queuing and execution time).  Two workflows can be running 
+    concurrently per repository.  1 virtual CPU.  Up to 3.75 GB of memory
+    https://developer.github.com/actions/creating-workflows/workflow-configuration-options/
+
+For Travis and Github actions, the compute limits are too restrictive to run the benchmarking in their environments.  
+Thus, we have applied for GCP and AWS credits to spin up cloud VMs to run the benchmark jobs.  The next question would 
```

Comment:
We haven't applied yet and there's no guarantee we will get them. Assume for now that we need to use a machine on a desk at USC as a build agent.